### PR TITLE
Add docs for the ImagePullPolicy configuration

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Documentation

### Description

This PR adds the documentation for the ImagePullPolicy configuration.

### Checklist

- [x] Update documentation

### CODE REVIEW

File: documentation/book/con-customizing-image-pull-policy.adoc [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+[id='con-customizing-image-pull-policy-{context}']
+= Customizing Image Pull Policy
+
+{ProductName} allows you to customize the Image Pull Policy for containers in all pods deployed by the Cluster Operator.
+Image Pull Policy is configured using the environment variable `STRIMZI_IMAGE_PULL_POLICY` in the Cluster Operator deployment.
+The `STRIMZI_IMAGE_PULL_POLICY` environment variable can be set to three different values:
+
+`Always`::
+Container images are pulled from the registry every time the the pod is started or restarted.
```
Comment (ppatierno) [inline documentation/book/con-customizing-image-pull-policy.adoc]: typo ... double "the"

File: documentation/book/con-customizing-image-pull-policy.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// This assembly is included in the following assemblies:
+//
+// assembly-customizing-deployments.adoc
+
+[id='con-customizing-image-pull-policy-{context}']
+= Customizing Image Pull Policy
```
Comment (ncbaratta) [inline documentation/book/con-customizing-image-pull-policy.adoc]:
```suggestion
= Customizing the image pull policy
```

File: documentation/book/con-customizing-image-pull-policy.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+{ProductName} allows you to customize the Image Pull Policy for containers in all pods deployed by the Cluster Operator.
```
Comment (ncbaratta) [inline documentation/book/con-customizing-image-pull-policy.adoc]: Why is Image Pull Policy in caps?

File: documentation/book/con-customizing-image-pull-policy.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Image Pull Policy is configured using the environment variable `STRIMZI_IMAGE_PULL_POLICY` in the Cluster Operator deployment.
```
Comment (ncbaratta) [inline documentation/book/con-customizing-image-pull-policy.adoc]:
```suggestion
The image pull policy is configured using the environment variable `STRIMZI_IMAGE_PULL_POLICY` in the Cluster Operator deployment.
```

File: documentation/book/con-customizing-image-pull-policy.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+`Always`::
+Container images are pulled from the registry every time the pod is started or restarted.
+
+`IfNotPresent`::
+Container images are pulled from the registry only when they were not pulled before.
+
+`Never`::
+Container images are never pulled from the registry.
+
+The Image Pull Policy can be currently customized only for all Kafka, Kafka Connect and Kafka Mirror Maker clusters at once.
```
Comment (ncbaratta) [inline documentation/book/con-customizing-image-pull-policy.adoc]:
```suggestion
The Image Pull Policy can be currently customized only for all Kafka, Kafka Connect, and Kafka Mirror Maker clusters at once.
```

File: documentation/book/con-customizing-image-pull-policy.adoc (hunk 2 above)
Comment (ncbaratta) [reply documentation/book/con-customizing-image-pull-policy.adoc]: Feel free to ignore suggestions about the capitalization of 'Image Pull Policy' if I'm wrong here.

File: documentation/book/con-customizing-image-pull-policy.adoc [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+Changing the policy will result in rolling update of all your Kafka, Kafka Connect and Kafka Mirror Maker clusters.
```
Comment (ncbaratta) [inline documentation/book/con-customizing-image-pull-policy.adoc]:
```suggestion
Changing the policy will result in a rolling update of all your Kafka, Kafka Connect, and Kafka Mirror Maker clusters.
```

File: documentation/book/con-customizing-image-pull-policy.adoc [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+.Additional resources
+
+* For more information about Cluster operator configuration, see xref:assembly-operators-cluster-operator-{context}[].
```
Comment (ncbaratta) [inline documentation/book/con-customizing-image-pull-policy.adoc]:
```suggestion
* For more information about Cluster Operator configuration, see xref:assembly-operators-cluster-operator-{context}[].
```

File: documentation/book/ref-operators-cluster-operator-configuration.adoc [hunk 8]
```diff
@@ -78,3 +78,9 @@ if no image is specified as the `Kafka.spec.entityOperator.userOperator.image` i
 `STRIMZI_DEFAULT_TLS_SIDECAR_ENTITY_OPERATOR_IMAGE`:: Optional, default `strimzi/entity-operator-stunnel:latest`.
 The image name to use as the default when deploying the sidecar container which provides TLS support for the Entity Operator, if
 no image is specified as the `Kafka.spec.entityOperator.tlsSidecar.image` in the xref:assembly-configuring-container-images-deployment-configuration-kafka[].
+
+`STRIMZI_IMAGE_PULL_POLICY`:: Optional.
+The `ImagePullPolicy` which will be applied to containers in all pods managed by {ProductName} Cluster Operator.
+The valid values are `Always`, `IfNotPresent` and `Never`.
```
Comment (ncbaratta) [inline documentation/book/ref-operators-cluster-operator-configuration.adoc]:
```suggestion
The valid values are `Always`, `IfNotPresent`, and `Never`.
```

File: documentation/book/ref-operators-cluster-operator-configuration.adoc [hunk 9]
```diff
@@ -78,3 +78,9 @@ if no image is specified as the `Kafka.spec.entityOperator.userOperator.image` i
...
+If not specified, the {ProductPlatformName} defaults will be used.
+Changing the policy will result in rolling update of all your Kafka, Kafka Connect and Kafka Mirror Maker clusters.
```
Comment (ncbaratta) [inline documentation/book/ref-operators-cluster-operator-configuration.adoc]:
```suggestion
Changing the policy will result in a rolling update of all your Kafka, Kafka Connect, and Kafka Mirror Maker clusters.
```

File: documentation/book/con-customizing-image-pull-policy.adoc (hunk 3 above)
Comment (scholzj) [reply documentation/book/con-customizing-image-pull-policy.adoc]: @ncbaratta Well, that is the usual question ... Image Pull Policy is a term from Kubernetes. Which always confuses me with how should I write it. I can change it all to lower case if you want.

Comment (ncbaratta) [reply documentation/book/con-customizing-image-pull-policy.adoc]: I think we should lowercase it - unless it's the variable name.

Comment (ncbaratta) [inline documentation/book/con-customizing-image-pull-policy.adoc]:
```suggestion
{ProductName} allows you to customize the image pull policy for containers in all pods deployed by the Cluster Operator.
```

File: documentation/book/con-customizing-image-pull-policy.adoc [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+`Always`::
+Container images are pulled from the registry every time the pod is started or restarted.
+
+`IfNotPresent`::
+Container images are pulled from the registry only when they were not pulled before.
+
+`Never`::
+Container images are never pulled from the registry.
+
+The Image Pull Policy can be currently customized only for all Kafka, Kafka Connect, and Kafka Mirror Maker clusters at once.
```
Comment (ncbaratta) [inline documentation/book/con-customizing-image-pull-policy.adoc]:
```suggestion
The image pull policy can be currently customized only for all Kafka, Kafka Connect, and Kafka Mirror Maker clusters at once.
```

File: documentation/book/con-customizing-image-pull-policy.adoc (hunk 3 above)
Comment (scholzj) [reply documentation/book/con-customizing-image-pull-policy.adoc]: Ok, lets change it to lower case then. Thanks.

**Pull request merged**